pkg/writer/sarif.go

```diff
@@ -65,19 +69,24 @@ func SarifWriter(data interface{}, writer io.Writer) error {
 		m["category"] = violation.Category
 		m["severity"] = violation.Severity
 
-		rule := run.AddRule(string(violation.RuleID)).
+		rule := run.AddRule(violation.RuleID).
 			WithDescription(violation.Description).WithName(violation.RuleName).WithProperties(m)
 
-		absFilePath, err := getAbsoluteFilePath(outputData.Summary.ResourcePath, violation.File)
-
-		if err != nil {
-			return err
+		var artifactLocation *sarif.ArtifactLocation
+
+		if forGithub {
+			artifactLocation = sarif.NewSimpleArtifactLocation(violation.File).
+				WithUriBaseId(outputData.Summary.ResourcePath)
+		} else {
+			absFilePath, err := getAbsoluteFilePath(outputData.Summary.ResourcePath, violation.File)
+			if err != nil {
+				return errors.Errorf("unable to create absolute path, error: %v", err)
```

Comment:
Is it possible to use `fmt.Errorf()` here?

Reply:
my bad. will correct this in future